### DISCUSSION TITLE
Next export alway need .html in URL

### DISCUSSION THREAD

**Discussion by yudijohn** [vercel/next.js#37654] [Help]
I tried to create a page /pages/test.html
then i run `next build && next export -o test`
after that, i move the test folder that builded by the next js to my production server
but when i try to access https://www.url.com/test, it's redirected to 404.html page
i try to access https://www.url.com/test.html, the page works

how to access it without .html in the url like what i access in dev mode?

**Comment by icyJoseph** [comment]
I think this has to do with how your server is handling requests. What type of architecture have you got there?

For example, loocally, you can install, https://www.npmjs.com/package/serve, and run `serve ./test` and that'll work out of the box. 

How is your production server configured? I'd search online about specific configurations to your server.

**Reply by yudijohn** [reply]
i'm running server with ubuntu server and nginx as the virtual host. so i can't run next js static HTML in my server?

**Reply by icyJoseph** [reply]
**Yes you can**, but you have to figure out how to tell the server to resolve `.html` extensions. Again, it is possible, but it depends on the server configs, this is something you have to learn. 

- https://stackoverflow.com/a/38238001
- https://serverfault.com/questions/346994/hide-html-file-extensions-using-nginx-rewrites
- https://gist.github.com/zackad/7dd46be60df4fe7d1010a0fcf33d1afc

Etc... there's specific knowledge to each server setup, perhaps there's already some configuration that's causing the issue, etc...
